tests/input/cpp/bug_1002.cpp

```diff
@@ -0,0 +1,4 @@
+template< class B1 = void, class B2 = void >
+struct conjunction : bool_constant<B1::value1 && B2::value2>
```

Comment:
I just noticed this, you are not testing if  `sp_bool = force` is actually doing anything here.
You could do this:

```cpp
struct conjunction : bool_constant<B1::value1  &&  B2::value2>
struct conjunction : bool_constant<B1::value1&&B2::value2>
```
To be sure there should also be a test with remove. The other two mentioned options should also be included to make sure that they are not causing misbehavior.